guides/keras_nlp/neural_machine_translation.py

```diff
@@ -0,0 +1,442 @@
+- Use the trained model to generate translations of unseen input sentences!
+- Vectorize text using the Keras `TextVectorization` layer.
+
+Don't worry if you aren't familiar with KerasNLP. This tutorial shows exactly
+how simple and easy it is to use it! So, what are you waiting for? Dive right in!
+"""
+
+"""
+## Setup
+Before we start implementing the pipeline, let's import all the libraries we need.
+"""
+
+"""shell
+pip install -q keras-nlp
+pip install -q -U "tensorflow-text==2.8.*"
+"""
+
+import keras_nlp
+import numpy as np
+import pathlib
+import random
+import tensorflow as tf
+
+from tensorflow import keras
+from tensorflow_text.tools.wordpiece_vocab import bert_vocab_from_dataset as bert_vocab
+
+"""
+Let's also define our hyperparameters.
+"""
+
+BATCH_SIZE = 64
+EPOCHS = 1  # Should be 20 to get decent results.
+MAX_SEQUENCE_LENGTH = 40
+ENG_VOCAB_SIZE = 15000
+SPA_VOCAB_SIZE = 15000
+
+EMBED_DIM = 256
+INTERMEDIATE_DIM = 2048
+NUM_HEADS = 8
+
+"""
+## Downloading the data
+
+We'll be working with an English-to-Spanish translation dataset
+provided by [Anki](https://www.manythings.org/anki/). Let's download it:
+"""
+
+text_file = keras.utils.get_file(
+    fname="spa-eng.zip",
+    origin="http://storage.googleapis.com/download.tensorflow.org/data/spa-eng.zip",
+    extract=True,
+)
+text_file = pathlib.Path(text_file).parent / "spa-eng" / "spa.txt"
+
+"""
+## Parsing the data
+
+Each line contains an English sentence and its corresponding Spanish sentence.
+The English sentence is the *source sequence* and Spanish one is the *target sequence*.
+Before adding the text to a list, we convert it to lowercase.
+"""
+
+with open(text_file) as f:
+    lines = f.read().split("\n")[:-1]
+text_pairs = []
+for line in lines:
+    eng, spa = line.split("\t")
+    eng = eng.lower()
+    spa = spa.lower()
+    text_pairs.append((eng, spa))
+
+"""
+Here's what our sentence pairs look like:
+"""
+
+for _ in range(5):
+    print(random.choice(text_pairs))
+
+"""
+Now, let's split the sentence pairs into a training set, a validation set,
+and a test set.
+"""
+
+random.shuffle(text_pairs)
+num_val_samples = int(0.15 * len(text_pairs))
+num_train_samples = len(text_pairs) - 2 * num_val_samples
+train_pairs = text_pairs[:num_train_samples]
+val_pairs = text_pairs[num_train_samples : num_train_samples + num_val_samples]
+test_pairs = text_pairs[num_train_samples + num_val_samples :]
+
+print(f"{len(text_pairs)} total pairs")
+print(f"{len(train_pairs)} training pairs")
+print(f"{len(val_pairs)} validation pairs")
+print(f"{len(test_pairs)} test pairs")
+
+
+"""
+## Tokenizing the data
+We'll define two tokenizers - one for the source language (English), and the other
+for the target language (Spanish). We'll be using the `WordPieceTokenizer` from
+KerasNLP to tokenize the text.
+
+
+Before we define the two tokenizers, we first need to train them on the dataset
+we have. WordPiece Tokenizer is a subword tokenizer; training it on a corpus gives
```

Comment:
Use backticks for code keywords, e.g. `WordPieceTokenizer`

Reply:
Actually, here, I am not referring to any code keyword. I am referring to WordPiece Tokenizer, and not the class `WordPieceTokenizer`.